.github/workflows/codespell.yaml

```diff
@@ -0,0 +1,12 @@
+name: codespell
+
+on: [push, pull_request]
+
+jobs:
+  codespell:
+    name: Codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: python3 -m pip install codespell
+      - run: codespell
```

Comment:
I like the idea of running codespell.

Please split the introduction of the tooling and the changes needed to make it happy into two commits.

Reply:
Yeah, i feared that it would have many false positives, but it actually had none.
I've now split the spell checking and tooling/CI changes into two commits.